twill-api/src/main/java/org/apache/twill/api/TwillController.java

```diff
@@ -89,4 +90,45 @@
    * @return A {@link Future} that will be completed when the restart operation has been done.
    */
   Future<String> restartInstances(String runnable, int instanceId, int... moreInstanceIds);
+
+  /**
+   * Set the log levels for requested logger names for Twill applications running in a container.
+   * The log level for a logger name can be {@code null} except for the root logger, which will reset the log level for
+   * the specified logger.
+   *
+   * @param logLevels The {@link Map} contains the requested logger names and log levels that need to be set.
+   * @return A {@link Future} that will be completed when the set log level operation has been done. It will carry the
+   *         {@link Map} of log levels as the result.
+   */
+  Future<Map<String, LogEntry.Level>> setLogLevels(Map<String, LogEntry.Level> logLevels);
+
+  /**
+   * Set the log levels for requested logger names for a {@link TwillRunnable}.
+   * The log level for a logger name can be {@code null} except for the root logger, which will reset the log level for
+   * the specified logger.
+   *
+   * @param runnableName The name of the runnable to set the log level.
+   * @param logLevelsForRunnable The {@link Map} contains the requested logger name and log level that
+   *                             need to be changed.
+   * @return A {@link Future} that will be completed when the set log level operation has been done. It will carry the
+   *         {@link Map} of log levels as the result.
+   */
+  Future<Map<String, LogEntry.Level>> setLogLevels(String runnableName,
+                                                   Map<String, LogEntry.Level> logLevelsForRunnable);
+
+  /**
+   * Reset the log levels of all runnables to the default log level, {@code LogEntry.Level.INFO}.
+   *
+   * @return A {@link Future} that will be completed when the set log level operation has been done.
+   */
+  Future<String> resetLogLevels();
+
+  /**
+   * Reset the log levels of the given runnable to the log levels when .
```

Comment:
My bad for not updating the comments. The `resetLogLevels` will reset the log level of a runnable to the time when it starts up. This means it will reset the log level to what `logback.xml` is + the log level requested from `TwillPreparer`